src/metachecking/Check_rule.ml

```diff
@@ -144,18 +144,26 @@ let unknown_metavar_in_comparison env f =
               pattern (or is only bound by negative patterns like \
               'pattern-not')")
         in
-        conditions
-        |> List.iter (fun (t, metavar_cond) ->
-               match metavar_cond with
-               | CondEval _ -> ()
-               | CondRegexp (mv, _, _) ->
-                   if not (mvar_is_ok mv mvs) then mv_error mv t
-               | CondType (mv, _, _, _) ->
-                   if not (mvar_is_ok mv mvs) then mv_error mv t
-               | CondNestedFormula (mv, _, _) ->
-                   if not (mvar_is_ok mv mvs) then mv_error mv t
-               | CondAnalysis (mv, _) ->
-                   if not (mvar_is_ok mv mvs) then mv_error mv t);
+        let cond_mvs =
```

Comment:
This is already a very large function, so better move this code in a separate helper function, like
mvars_in_metavar_cond